src/grammar/trigger.rs

```diff
@@ -0,0 +1,656 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE.md
+// SPDX-FileCopyrightText: 2023 CYBERTEC PostgreSQL International GmbH
+// <office@cybertec.at>
+
+use crate::lexer::TokenKind;
+use crate::parser::Parser;
+use crate::syntax::SyntaxKind;
+
+use super::*;
+
+/// Parses a complete trigger.
+pub(crate) fn parse_trigger(p: &mut Parser) {
+    p.start(SyntaxKind::Trigger);
+    parse_header(p);
+    parse_body(p);
+    p.finish();
+}
+
+/// Parses the header of a trigger.
+fn parse_header(p: &mut Parser) {
+    p.start(SyntaxKind::TriggerHeader);
+    p.expect(T![create]);
+    if p.eat(T![or]) {
+        p.expect(T![replace]);
+    }
+
+    p.eat_one_of(&[T![editionable], T![noneditionable]]);
+
+    p.expect(T![trigger]);
+    parse_ident(p, 1..2);
+
+    match p.current() {
+        T![before] | T![instead] | T![after] => {
+            if p.eat(T![instead]) {
+                p.expect(T![of]);
+            } else {
+                p.bump_any();
+            }
+
+            match p.current() {
+                T![insert] | T![update] | T![delete] => parse_simple_dml_trigger(p),
+                _ => parse_system_trigger(p),
+            };
+        }
+        T![for] => {
+            p.error(ParseError::Unimplemented("compound trigger".to_string()));
+        }
+        _ => {
+            p.expect_one_of(&[T![before], T![instead], T![after], T![for]]);
+        }
+    }
+
+    p.finish();
+}
+
+fn parse_simple_dml_trigger(p: &mut Parser) {
+    parse_dml_event_clause(p);
+    parse_referencing_clause(p);
+
+    if p.eat(T![for]) {
+        p.expect(T![each]);
+        p.expect(T![row]);
+    }
+
+    parse_trigger_edition_clause(p);
+    parse_trigger_ordering_clause(p);
+
+    p.eat_one_of(&[T![enable], T![disable]]);
+
+    if p.eat(T![when]) {
+        p.expect(T!["("]);
+        parse_expr(p);
+        p.expect(T![")"]);
+    }
+}
+
+fn parse_system_trigger(p: &mut Parser) {
+    loop {
+        let bump_n = match &[&[p.current()], p.lookahead(2).as_slice()].concat()[..] {
```

Comment:
The `match` statement is clearer now, I like it.

I have only a very small suggestion:

```rust
match [p.current(), p.nth(1), p.nth(2)] {}
```